src/Hazelcast.Net/Messaging/ClientMessageConnection.cs

```diff
@@ -296,7 +296,15 @@ public async ValueTask<bool> SendAsync(ClientMessage message, CancellationToken
             }
             finally
             {
-                _writer?.Release();
```

Comment:
What about adding a `disposed` boolean variable to HSemaphore? See https://learn.microsoft.com/en-us/dotnet/api/system.idisposable.dispose?view=net-6.0 for more.

Reply:
The `IHSemaphore` interface exists only to wrap a `SemaphoreSlim` instance in production, but replace it and mock it during tests. So we want it to behave exactly as `SemaphoreSlim` does. And, `SemaphoreSlim` throws if you release it once disposed. We try to keep that very same behavior.

Reply:
Ok. Approved.